owncloudApp/src/main/java/com/owncloud/android/providers/DocumentsStorageProvider.kt

```diff
@@ -0,0 +1,300 @@
+/**
+ * ownCloud Android client application
+ *
+ * @author Bartosz Przybylski
+ * @author Christian Schabesberger
+ * @author David González Verdugo
+ * @author Abel García de Prada
+ * Copyright (C) 2015  Bartosz Przybylski
+ * Copyright (C) 2019 ownCloud GmbH.
+ * <p>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ * <p>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.providers
+
+import android.annotation.TargetApi
+import android.content.Intent
+import android.content.res.AssetFileDescriptor
+import android.database.Cursor
+import android.graphics.Point
+import android.net.Uri
+import android.os.CancellationSignal
+import android.os.ParcelFileDescriptor
+import android.provider.DocumentsContract
+import android.provider.DocumentsProvider
+import com.owncloud.android.R
+import com.owncloud.android.authentication.AccountUtils
+import com.owncloud.android.datamodel.FileDataStorageManager
+import com.owncloud.android.datamodel.OCFile
+import com.owncloud.android.files.services.FileDownloader
+import com.owncloud.android.lib.common.utils.Log_OC
+import com.owncloud.android.operations.RefreshFolderOperation
+import com.owncloud.android.operations.RenameFileOperation
+import com.owncloud.android.providers.cursors.FileCursor
+import com.owncloud.android.providers.cursors.RootCursor
+import java.io.File
+import java.io.FileNotFoundException
+import java.util.HashMap
+import java.util.Vector
+
+class DocumentsStorageProvider : DocumentsProvider() {
+    /**
+     * If a directory requires to sync, it will write the id of the directory into this variable.
+     * After the sync function gets triggered again over the same directory, it will see that a sync got already
+     * triggered, and does not need to be triggered again. This way a endless loop is prevented.
+     */
+    private var requestedFolderIdForSync: Long = -1
```

Comment:
This variable is only used in queryChildDocuments so it does not need to be a global variable.

Reply:
But we need it for next calls